### PR TITLE
Use Proper Camel Casing with Acronyms

### LINKED ISSUE #5518: Follow CamelCasing in Method Signatures that use Acronyms
Change acronyms to use camel case for these methods

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes #5518 

Updated API methods that had uppercased acronyms to follow proper camel casing.

Updating blob listing methods to align with .NET (listBlobs and listBlobsByHierarchy).

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java [hunk 1]
```diff
@@ -467,9 +467,9 @@ public URL getAccountUrl() {
      * @param expiryTime The {@code OffsetDateTime} expiry time for the account SAS
      * @return A string that represents the SAS token
      */
-    public String generateAccountSAS(AccountSASService accountSASService, AccountSASResourceType accountSASResourceType,
+    public String generateAccountSas(AccountSASService accountSASService, AccountSASResourceType accountSASResourceType,
```
Comment (conniey) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java]: Your parameter names have the same casing as before.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java [hunk 2]
```diff
@@ -743,12 +743,12 @@ private void downloadToFileCleanup(AsynchronousFileChannel channel) {
      * @param accessConditions {@link BlobAccessConditions}
      * @return A reactive response signalling completion.
      */
-    public Mono<Response<Void>> setHTTPHeadersWithResponse(BlobHTTPHeaders headers,
+    public Mono<Response<Void>> setHttpHeadersWithResponse(BlobHTTPHeaders headers,
```
Comment (conniey) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java]: Is the BlobHTTPHeaders class going to change too?

File: sdk/storage/azure-storage-queue/src/samples/java/com/azure/storage/queue/QueueJavaDocCodeSamples.java [hunk 3]
```diff
@@ -444,8 +444,7 @@ public void clearMessagesWithResponse() {
     }
 
     /**
-     * Code snippet for {@link QueueClient#generateSAS(String, QueueSASPermission, OffsetDateTime, OffsetDateTime,
-     * String, SASProtocol, IPRange)}
+     * Code snippet for {@link QueueClient#generateSas(String, QueueSASPermission, OffsetDateTime, OffsetDateTime, String, SASProtocol, IPRange)}
```
Comment (conniey) [inline sdk/storage/azure-storage-queue/src/samples/java/com/azure/storage/queue/QueueJavaDocCodeSamples.java]: Not sure if you own the model class generation, (ie. QueueSASPermission and SASProtocol) and if that is going to be changed.

File: sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueServiceClient.java [hunk 4]
```diff
@@ -402,10 +402,10 @@ public StorageServiceStats getStatistics() {
      * @param expiryTime The {@code OffsetDateTime} expiry time for the account SAS
      * @return A string that represents the SAS token
      */
-    public String generateAccountSAS(AccountSASService accountSASService,
+    public String generateAccountSas(AccountSASService accountSASService,
```
Comment (conniey) [inline sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueServiceClient.java]: parameter names here should be fixed?

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java]: This is a codegen class, not sure how possible it is to fix this casing

Comment (conniey) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java]: @jianghaolu Is this something that can be fixed in autorest?

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BaseBlobClientBuilder.java [hunk 5]
```diff
@@ -3,39 +3,68 @@
...
 import com.azure.core.http.policy.UserAgentPolicy;
 import com.azure.storage.blob.models.CpkInfo;
 import com.azure.storage.blob.models.CustomerProvidedKey;
 import com.azure.storage.common.BaseClientBuilder;
 import com.azure.storage.common.Constants;
 import com.azure.storage.common.policy.ResponseValidationPolicyBuilder;
 
-abstract class BaseBlobClientBuilder<T extends BaseClientBuilder<T>> extends BaseClientBuilder<T> {
+/**
+ * Base builder for Azure Storage Blobs.
```
Comment (conniey) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BaseBlobClientBuilder.java]: nit: there's a blank line between summary and params in your other methods. I'd add one here.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobProperties.java [hunk 6]
```diff
@@ -49,7 +49,7 @@
     private final Metadata metadata;
     private final Integer committedBlockCount;
 
-    BlobProperties(BlobGetPropertiesHeaders generatedHeaders) {
+    public BlobProperties(BlobGetPropertiesHeaders generatedHeaders) {
```
Comment (conniey) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobProperties.java]: Do you expect people to create this class? I'm curious why it is public. Missing documentation if you intend to keep it public.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobUrlParts.java [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+     * @param scheme The URL scheme.
+     * @return the updated BlobUrlParts object.
+     */
+    public BlobUrlParts setScheme(String scheme) {
+        this.scheme = scheme;
+        return this;
+    }
+
+    /**
+     * Gets the URL host, ex. "account.blob.core.windows.net".
```
Comment (conniey) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobUrlParts.java]: What you're returning is the hostname, not a host. The documentation is misleading.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BaseBlobClientBuilder.java [hunk 8]
```diff
@@ -3,39 +3,68 @@
...
+            customerProvidedKey = null;
+        } else {
+            customerProvidedKey = new CpkInfo()
+                .setEncryptionKey(key.getKey())
+                .setEncryptionKeySha256(key.getKeySHA256())
+                .setEncryptionAlgorithm(key.getEncryptionAlgorithm());
+        }
 
-        return (T) this;
+        return getClazz().cast(this);
```
Comment (JonathanGiles) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BaseBlobClientBuilder.java]: Huh?

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BaseBlobClientBuilder.java [hunk 9]
```diff
@@ -3,39 +3,68 @@
...
 import com.azure.storage.common.BaseClientBuilder;
 import com.azure.storage.common.Constants;
 import com.azure.storage.common.policy.ResponseValidationPolicyBuilder;
 
-abstract class BaseBlobClientBuilder<T extends BaseClientBuilder<T>> extends BaseClientBuilder<T> {
+/**
+ * Base builder for Azure Storage Blobs.
+ * @param <T> Generic type that extends {@link BaseClientBuilder}.
+ */
+public abstract class BaseBlobClientBuilder<T extends BaseClientBuilder<T>> extends BaseClientBuilder<T> {
```
Comment (JonathanGiles) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BaseBlobClientBuilder.java]: Not a big fan of this build inheritance hierarchy. We should take a closer look at this next week.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/ContainerClientBuilder.java [hunk 10]
```diff
@@ -124,6 +124,11 @@ public ContainerClientBuilder containerName(String containerName) {
         return this;
     }
 
+    @Override
+    protected Class<ContainerClientBuilder> getClazz() {
+        return ContainerClientBuilder.class;
+    }
```
Comment (JonathanGiles) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/ContainerClientBuilder.java]: Not liking this method at all. It should be removed.

**Pull request closed without merging**